pkg/errors/error.go

```diff
@@ -13,15 +14,56 @@ import (
 var (
 	New    = errors.New
 	Unwrap = errors.Unwrap
-	Is     = errors.Is
-	As     = errors.As
 )
 
 func Newf(msg string, args ...interface{}) error {
 	return New(fmt.Sprintf(msg, args...))
 }
 
+func As(err error, target any) bool {
+	if errors.As(err, target) {
+		return true
+	}
+
+	for err != nil {
+		if list, ok := err.(*ErrorList); ok {
+			for _, n := range list.errors {
+				if As(n, target) {
+					return true
+				}
+			}
+		}
+		err = Unwrap(err)
+	}
+	return false
+}
+
+func Is(err error, target error) bool {
```

Comment:
Any particular reason why Go's errors.Is and errors.As is reimplemented here but slightly differently?

Reply:
The original `Is`and `As` functions do not care about error lists. This is what the new implementations add.